### PR TITLE
Add support for TuYa-TS130F/LoraTap-SC400 Curt-SW

### PR DESCRIPTION
**Pull request by 1subu**
I added support for the TuYa-TS130F/LoraTap-SC400 Curtain/RollerShutter Switch. Discussed in https://github.com/Koenkk/zigbee2mqtt/issues/4257

### CODE REVIEW

File: converters/fromZigbee.js [hunk 1]
```diff
@@ -1391,6 +1391,39 @@ const converters = {
...
+            return result;
+        },
+    },
+    tuya_backlight_mode: {
+        cluster: 'genOnOff',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            if (msg.data.hasOwnProperty('tuyaBacklightMode')) {
+                const value = msg.data['tuyaBacklightMode'];
+                return {backlight_mode: value};
```
Comment (Koenkk) [inline converters/fromZigbee.js:1423]: Can you convert value to `ON`, `OFF` here, this will now return 1/0.

File: converters/fromZigbee.js [hunk 2]
```diff
@@ -1391,6 +1391,39 @@ const converters = {
...
         },
     },
+    tuya_cover_options: {
+        cluster: 'closuresWindowCovering',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const result = {};
+            if (msg.data.hasOwnProperty('tuyaMovingState')) {
+                const value = msg.data['tuyaMovingState'];
+                const movingLookup = {0: 'up', 1: 'stop', 2: 'down'};
```
Comment (Koenkk) [inline converters/fromZigbee.js:1401]: Please make `UP`, `STOP`, `DOWN` and `ON`/`OFF` below upper case.

File: converters/toZigbee.js [hunk 3]
```diff
@@ -3441,7 +3441,40 @@ const converters = {
             );
         },
     },
-
+    tuya_cover_calibration: {
+        key: ['calibration'],
+        convertSet: async (entity, key, value, meta) => {
+            const lookup = {'on': 0, 'off': 1};
```
Comment (Koenkk) [inline converters/toZigbee.js:3447]: Also make on/off upper case here.

File: converters/toZigbee.js [hunk 4]
```diff
@@ -3441,7 +3441,40 @@ const converters = {
...
+            return {state: {calibration: value}};
+        },
+        convertGet: async (entity, key, meta) => {
+            await entity.read('closuresWindowCovering', ['tuyaCalibration']);
+        },
+    },
+    tuya_cover_reversal: {
+        key: ['motor_reversal'],
+        convertSet: async (entity, key, value, meta) => {
+            const lookup = {'on': 1, 'off': 0};
```
Comment (Koenkk) [inline converters/toZigbee.js:3459]: Also make on/off upper case here.

File: converters/toZigbee.js [hunk 5]
```diff
@@ -3441,7 +3441,40 @@ const converters = {
...
+        },
+        convertGet: async (entity, key, meta) => {
+            await entity.read('closuresWindowCovering', ['tuyaMotorReversal']);
+        },
+    },
+    tuya_backlight_mode: {
+        key: ['backlight_mode'],
+        convertSet: async (entity, key, value, meta) => {
+            await entity.write('genOnOff', {tuyaBacklightMode: value});
+            return {state: {backlight_mode: value}};
```
Comment (Koenkk) [inline converters/toZigbee.js:3472]: Allow on/off here.

File: devices.js [hunk 6]
```diff
@@ -14001,6 +14001,20 @@ const devices = [
...
         extend: generic.light_onoff_brightness,
     },
+	// LoraTap SC400 curtain switch
+    {
+        zigbeeModel: ['TS130F'],
+		fingerprint: [{modelID: 'TS130F', manufacturerName: '_TZ3000_8kzqqzu4'}],
+        model: 'TS0601_curtain_switch',
+        vendor: 'TuYa',
+        description: 'Curtain/Blind switch',
+        supports: 'open, close, stop, position',
```
Comment (Koenkk) [inline devices.js:14011]: Please add `exposes` here (for automatic home assistant discovery and control via frontend: https://www.zigbee2mqtt.io/information/frontend.html). 

The following should work (but please test):

```
exposes: [e.cover_position(), exposes.binary('calibration', 'rw', 'ON', 'OFF'), exposes.binary('backlight_mode', 'rw', 'ON', 'OFF'), exposes.binary('motor_reversal', 'rw', 'ON', 'OFF')],
```

File: converters/fromZigbee.js (hunk 1 above)
Comment (1subu) [reply converters/fromZigbee.js:1423]: Tuya has 3 lightmodes. Values are ('0', '1', '2'). I cannot convert it to ON OFF. I would leave the original value.

File: converters/toZigbee.js (hunk 5 above)
Comment (1subu) [reply converters/toZigbee.js:3472]: Tuya has 3 lightmodes. Values are ('0', '1', '2'). I will check to these values. OK?

File: devices.js (hunk 6 above)
Comment (1subu) [reply devices.js:14011]: calibration and motor_reversal seems to work.
For backlight_mode i would use: 
exposes.enum('backlight_mode', 'rw', ['0', '1', '2']),

File: converters/fromZigbee.js (hunk 2 above)
Comment (1subu) [reply converters/fromZigbee.js]: Done

File: converters/toZigbee.js (hunk 3 above)
Comment (1subu) [reply converters/toZigbee.js]: Done

File: converters/toZigbee.js (hunk 4 above)
Comment (1subu) [reply converters/toZigbee.js]: Done

File: converters/fromZigbee.js (hunk 1 above)
Comment (Koenkk) [reply converters/fromZigbee.js:1423]: In that case use `low`, `medium`, `high`.

Comment (1subu) [reply converters/fromZigbee.js]: done

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**